File: Class Representation/Association/C++/Association.cpp
#include <iostream>
#include <vector>
using namespace std;

// Unidirectional Association
class PaymentGateway
{
public:
    void processPayment(double amount)
    {
        cout << "Processing payment of $" << amount << endl;
    }
};

class Order
{
private:
    PaymentGateway *paymentgateway;

public:
    Order(PaymentGateway *paymentgateway)
    {
        this->paymentgateway = paymentgateway;
    }

    void checkout()
    {
        paymentgateway->processPayment(100.0);
    }
};

// Bidirectional Association
class Developer
{

    private:
        Team* team;

    public:
        void setTeam(Team* team){
            this->team = team;
        }
};

class Team
{
    private:
        vector<Developer*> devs;

    public:
        void addDevs(Developer* dev){
            devs.push_back(dev);
            dev->setTeam(this);
        }
};

int main()
{

    return 0;
}

File: Class Representation/Association/Typescript/Association.ts
// Unidirectional Association
class PaymentGateway {
  paymentProcess(amount: number): void {
    console.log(`Processing payment of $${amount}`);
  }
}

class Order {
  private gatway: PaymentGateway;

  constructor(paymentgateway: PaymentGateway) {
    this.gatway = paymentgateway;
  }
  checkOut(): void {
    this.gatway.paymentProcess(100.0);
  }
}

// Bidirectional Association

class Developer {
  private team?: Team;

  setTeam(team: Team): void {
    this.team = team;
  }
}

class Team {
  private devs: Developer[] = [];

  addDevs(dev: Developer): void {
    this.devs.push(dev);
    dev.setTeam(this);
  }
}
